ql/src/java/org/apache/hadoop/hive/ql/ddl/table/create/CreateTableDesc.java

```diff
@@ -937,7 +936,9 @@ public Table toTable(HiveConf conf) throws HiveException {
             tbl.getTTable().getDictionary() : new ObjectDictionary();
         List<ByteBuffer> buffers = new ArrayList<>();
         String statsSetup = StatsSetupConst.ColumnStatsSetup.getStatsSetupAsString(true,
-            storageHandler != null && storageHandler.isMetadataTableSupported() ? "metadata" : null, // Skip metadata directory for Iceberg table
+            // Ignore all Iceberg leftover files when storageHandler.canProvideBasicStatistics() is true,
+            // as the method is only enabled in Iceberg currently.
+            storageHandler != null && storageHandler.canProvideBasicStatistics(),
```

Comment:
@dengzhhu653  Sorry, I didn't notice that you were using 'isMetadataTableSupported' to determine if the table is Iceberg, but not that it can provide column stats.
`canProvideBasicStatistics` won't be the proper way for that check. 'isMetadataTableSupported' would be a better option here.

Reply:
Ok, I'll restore the method